src/microbe_stage/Microbe.Interior.cs

```diff
@@ -378,6 +378,34 @@ public void SetInitialCompounds()
         {
             Compounds.AddCompound(entry.Key, entry.Value);
         }
+
+        var setting = GameWorld.WorldSettings;
+
+        // Fraction of day that is night
+        var nightTimeFraction = 1 - setting.DaytimeFraction;
+
+        // Shifts fraction so the day ends exactly 1 (ignoring padding).
+        var shiftedDay = (GameWorld.LightCycle.FractionOfDayElapsed + nightTimeFraction / 2 +
+            Constants.INITIAL_COMPOUND_EVENING_PADDING_FRACTION) % 1;
+
+        if (shiftedDay < nightTimeFraction)
+        {
+            // This should never trigger, but we need to assure the linter that
+            // organelles is never null here.
+            if (organelles == null)
+                throw new ArgumentNullException();
```

Comment:
I don't think a cache works, as this is a dynamic value based on the time of day.

Reply:
You mean the calculated compound balance changes? Still from small lag spike viewpoint it's not a good design. I commented later that some kind of more in-depth check could be made to calculate which compounds vary during the night. I see that as potentially much more useful calculation result that can be cached and from it it could be just a quick calculation to determine the resources to give to each spawned cell.